### PR TITLE
[Documentation] Add Symbol Documentation in Source Files

### PR DESCRIPTION
**Pull request by AmyL219**
## Purpose

* In order to generate documentation based on [Jazzy](https://github.com/Azure/azure-sdk-for-ios/wiki/Publishing-Jazzy-Docs-to-GitHub.io), add documentation in the source files
* The [preview draft](https://htmlpreview.github.io/?https://raw.githubusercontent.com/Azure/azure-sdk-for-ios/communication/CommunicationUIJazzy/build/jazzy/AzureCommunicationUI/index.html) of the generated documents. Notice: Be free to notice me if there is any error in the doc, like the version needs a update, etc

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUI/CallComposite.swift [hunk 1]
```diff
@@ -17,10 +18,14 @@ public class CallComposite {
     private var permissionManager: AppPermissionsManager?
     private var audioSessionManager: AppAudioSessionManager?
 
+    /// Create an instance of CallComposite with options.
+    /// - Parameter options: The CallCompositeOptions used to configure the experience.
     public init(withOptions options: CallCompositeOptions) {
         themeConfiguration = options.themeConfiguration
     }
 
+    /// Assign an action to perform when an error occurs.
```
Comment (JoshuaLai) [inline AzureCommunicationUI/AzureCommunicationUI/CallComposite.swift:27]:
```suggestion
    /// Assign closure to execute if / when the composite encounters an error.
```

File: AzureCommunicationUI/AzureCommunicationUI/CallComposite.swift [hunk 2]
```diff
@@ -17,10 +18,14 @@ public class CallComposite {
...
+    /// - Parameter action: The closure on subscribing the error thrown from Call Composite.
```
Comment (JoshuaLai) [inline AzureCommunicationUI/AzureCommunicationUI/CallComposite.swift:28]:
```suggestion
    /// - Parameter action: The closure returning the error thrown from Call Composite.
```

Comment (JoshuaLai) [reply AzureCommunicationUI/AzureCommunicationUI/CallComposite.swift:28]: nit: Just as a side note the format is just a little different here than in other files. 
```
/// - Parameter:
///      action: The closure on subscribing the error thrown from Call Composite.
```

**Comment by JoshuaLai** [comment]
Had just some minor suggestions otherwise the jazzy generated documents look good.

File: AzureCommunicationUI/AzureCommunicationUI/CallComposite.swift (hunk 2 above)
Comment (AmyL219) [reply AzureCommunicationUI/AzureCommunicationUI/CallComposite.swift:28]: I use `cmd`+`opt`+G to automatically generate those format from Apple. I guess Apple has different format when there is only one parameter vs multiple parameters

**Review by kevinyulu — APPROVED** [review]

**Review by jimchou-dev — APPROVED** [review]

**Review by JoshuaLai — APPROVED** [review]

**Pull request merged**